WcaOnRails/app/webpacker/components/EditSchedule/EditActivities/index.js

```diff
@@ -0,0 +1,443 @@
+import React, {
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  Button,
+  Container,
+  Divider,
+  Dropdown,
+  Form,
+  Grid,
+  Icon,
+  Message,
+  Popup, Ref, Segment,
+  Sticky,
+} from 'semantic-ui-react';
+
+import FullCalendar from '@fullcalendar/react';
+import interactionPlugin, { Draggable } from '@fullcalendar/interaction';
+import timeGridPlugin from '@fullcalendar/timegrid';
+import luxonPlugin, { toLuxonDateTime, toLuxonDuration } from '@fullcalendar/luxon3';
+
+import { useDispatch, useStore } from '../../../lib/providers/StoreProvider';
+import { useConfirm } from '../../../lib/providers/ConfirmProvider';
+import useInputState from '../../../lib/hooks/useInputState';
+import ActivityPicker from './ActivityPicker';
+import { roomWcifFromId, venueWcifFromRoomId } from '../../../lib/utils/wcif';
+import { getTextColor } from '../../../lib/utils/calendar';
+
+import {
+  addActivity,
+  editActivity,
+  moveActivity,
+  removeActivity,
+  scaleActivity,
+} from '../store/actions';
+
+import { friendlyTimezoneName } from '../../../lib/wca-data.js.erb';
+import {
+  defaultDurationFromActivityCode,
+  fcEventToActivityAndDates,
+  luxonToWcifIso,
+} from '../../../lib/utils/edit-schedule';
+import EditActivityModal from './EditActivityModal';
+
+function EditActivities({
+  wcifEvents,
+  calendarLocale,
+}) {
+  const { wcifSchedule } = useStore();
+  const dispatch = useDispatch();
+
+  const confirm = useConfirm();
+
+  const [selectedRoomId, setSelectedRoomId] = useInputState();
+
+  const [minutesPerRow, setMinutesPerRow] = useInputState(15);
+  const [calendarStart, setCalendarStart] = useInputState(8);
+  const [calendarEnd, setCalendarEnd] = useInputState(20);
+
+  // This part is ugly because Semantic-UI and Fullcalendar disagree
+  //   about how modals should be handled.
+  // According to Semantic-UI, modals are "always there" in the DOM,
+  //   just that their "isOpen" state is false most of the time.
+  //   So they simply don't show but they are already part of the DOM tree.
+  // The (click-)event-based model of Fullcalendar however dictates that
+  //   we can only "instantiate" a modal once the user actually clicks somewhere on the calendar.
+  //   So we pre-fill the modal with empty state and set it accordingly on every event click.
+  // If somebody has a better idea how to handle this, please shout.
```

Comment:
I had a similar issue once. At first we tried only including the modal when it was open:
```
...
  {showActivityModal && <EditActivityModal
                                          // showModal={showActivityModal} // <- don't need this anymore
                                          activity={modalActivity}
                                          startLuxon={modalLuxonStart}
                                          endLuxon={modalLuxonEnd}
                                          dateLocale={calendarLocale}
                                          onModalClose={onActivityModalClose}
                                        />}
...
```
This worked well for having the props/data behave as expected, but it ruined the open/close animation on the modal so ultimately we switched to more-or-less what you're current doing.

Reply:
Okay, thanks for hte input! In terms of code changes, I will consider this comment a no-op